synctest/run_app.go

```diff
@@ -0,0 +1,244 @@
+package synctest
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/celestiaorg/test-infra/testkit"
+	"github.com/celestiaorg/test-infra/testkit/appkit"
+	"github.com/testground/sdk-go/network"
+	"github.com/testground/sdk-go/run"
+	"github.com/testground/sdk-go/runtime"
+)
+
+func RunAppValidator(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
```

Comment:
This func should be DRYied together with the results from #27. It's like 95% of repetitive code here. Ideally, the validator setup (this code) should be split into multiple sub-operations so we can compose them. I would do it earlier than later but surely can be a separate refactoring issue.

Reply:
Thank you so much for taking your time and deep diving into the details 🙏 
This should be resolved as we get #46 resolved